controller/recipeController.js

```diff
@@ -31,17 +32,89 @@ const getRecipes = async (req, res) => {
         if (!user_id) {
             return res.status(400).json({ error: 'User Id is required', statusCode: 400 });
         }
+        let recipeList = []
+        let cuisineList = []
+        let ingredientList = []
 
-        const recipes = await getUserRecipes(user_id);
+        const recipeRelation = await getUserRecipes.getUserRecipesRelation(user_id);
+        if (recipeRelation.length === 0) {
+            return res.status(404).json({ error: 'Recipes not found', statusCode: 404 });
+        }
+
+        for (let i = 0; i < recipeRelation.length; i++) {
+            if (i === 0) {
+                recipeList.push(recipeRelation[i].recipe_id);
+                cuisineList.push(recipeRelation[i].cuisine_id)
+                ingredientList.push(recipeRelation[i].ingredient_id)
+            }
+            else if (recipeList.indexOf(recipeRelation[i].recipe_id) < 0) {
+                recipeList.push(recipeRelation[i].recipe_id);
+            }
+            else if (cuisineList.indexOf(recipeRelation[i].cuisine_id) < 0) {
+                cuisineList.push(recipeRelation[i].cuisine_id)
+            }
+            else if (ingredientList.indexOf(recipeRelation[i].ingredient_id) < 0) {
+                ingredientList.push(recipeRelation[i].ingredient_id)
+            }
+        };
+
+        const recipes = await getUserRecipes.getUserRecipes(recipeList)
         if (recipes.length === 0) {
             return res.status(404).json({ error: 'Recipes not found', statusCode: 404 });
         }
 
+        const ingredients = await getUserRecipes.getIngredients(ingredientList)
+        if (ingredients.length === 0) {
+            return res.status(404).json({ error: 'Ingredients not found', statusCode: 404 });
+        }
+
+        const cuisines = await getUserRecipes.getCuisines(cuisineList)
+        if (cuisines.length === 0) {
+            return res.status(404).json({ error: 'Cuisines not found', statusCode: 404 });
+        }
+
+        recipes.forEach(function (recipe) {
```

Comment:
This isnt a deal breaker but I think this code could be improved by making it immutable rather than rely on modifying the recipe object we should make a copy and modify it as required.

Reply:
I'm not modifying recipe object, only adding string label arrays/values due to sending only IDs could be challenging for frontend to support config in each ingredients/recipe interaction.

that part could be deleted in future, I was thinking about deleting it now, but remained for a while